mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageMapper.java

```diff
@@ -228,13 +226,22 @@ public MessageUid findFirstUnseenMessageUid(Mailbox mailbox) throws MailboxExcep
 
     private CompletableFuture<Stream<SimpleMailboxMessage>> expungeUidChunk(CassandraId mailboxId, Collection<MessageUid> uidChunk) {
         return FluentFutureStream.ofOptionals(
-                uidChunk.stream().map(uid -> messageIdDAO.retrieve(mailboxId, uid)))
+                uidChunk.stream().map(uid -> retrieveComposedId(mailboxId, uid)))
             .performOnAll(this::deleteUsingMailboxId)
             .thenFlatCompose(idWithMetadata -> retrieveMessagesAndDoMigrationIfNeeded(ImmutableList.of(idWithMetadata), FetchType.Metadata, Limit.unlimited()))
             .map(pair -> pair.getKey().toMailboxMessage(ImmutableList.of()))
             .completableFuture();
     }
 
+    private CompletableFuture<Optional<ComposedMessageIdWithMetaData>> retrieveComposedId(CassandraId mailboxId, MessageUid uid) {
+        return messageIdDAO.retrieve(mailboxId, uid)
+            .thenApply(value -> Optional.of(value)
+            .orElseGet(() -> {
+                LOGGER.warn("Could not retrieve message {} {}. This might be due to denormalisation issues.", mailboxId, uid);
```

Comment:
please don't include the "might be ..." because it might also be something else ...